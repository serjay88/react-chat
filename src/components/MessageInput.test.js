/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from './MessageInput';

const mockData = {
  showJoinButton: true,
  onJoinButtonClick: jest.fn(),
  sendMessage: jest.fn(),
  disabled: false,
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...mockData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders input ', () => {
    const tree = renderer.create(<MessageInput {...mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders disabled input ', () => {
    const tree = renderer.create(<MessageInput {...mockData} disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders button ', () => {
    const tree = renderer.create(<MessageInput {...mockData} showJoinButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders disabled button', () => {
    const tree = renderer.create(<MessageInput {...mockData} showJoinButton disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
