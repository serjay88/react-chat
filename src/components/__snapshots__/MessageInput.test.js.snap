// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders button  1`] = `
<div
  className="MessageInput-messageInputWrapper-55"
>
  <div
    className="MuiPaper-root-57 MuiPaper-elevation6-65 MuiPaper-rounded-58 MessageInput-messageInput-56"
  >
    <button
      className="MuiButtonBase-root-99 MuiButton-root-84 MuiButton-raised-89 MuiButton-raisedPrimary-91 MuiButton-fullWidth-98"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-85"
      >
        Join
      </span>
      <span
        className="MuiTouchRipple-root-102"
      />
    </button>
  </div>
</div>
`;

exports[`<MessageInput /> renders disabled button 1`] = `
<div
  className="MessageInput-messageInputWrapper-55"
>
  <div
    className="MuiPaper-root-57 MuiPaper-elevation6-65 MuiPaper-rounded-58 MessageInput-messageInput-56"
  >
    <button
      className="MuiButtonBase-root-99 MuiButtonBase-disabled-100 MuiButton-root-84 MuiButton-raised-89 MuiButton-raisedPrimary-91 MuiButton-disabled-93 MuiButton-fullWidth-98"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiButton-label-85"
      >
        Join
      </span>
    </button>
  </div>
</div>
`;

exports[`<MessageInput /> renders disabled input  1`] = `
<div
  className="MessageInput-messageInputWrapper-55"
>
  <div
    className="MuiPaper-root-57 MuiPaper-elevation6-65 MuiPaper-rounded-58 MessageInput-messageInput-56"
  >
    <button
      className="MuiButtonBase-root-99 MuiButtonBase-disabled-100 MuiButton-root-84 MuiButton-raised-89 MuiButton-raisedPrimary-91 MuiButton-disabled-93 MuiButton-fullWidth-98"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiButton-label-85"
      >
        Join
      </span>
    </button>
  </div>
</div>
`;

exports[`<MessageInput /> renders input  1`] = `
<div
  className="MessageInput-messageInputWrapper-55"
>
  <div
    className="MuiPaper-root-57 MuiPaper-elevation6-65 MuiPaper-rounded-58 MessageInput-messageInput-56"
  >
    <button
      className="MuiButtonBase-root-99 MuiButton-root-84 MuiButton-raised-89 MuiButton-raisedPrimary-91 MuiButton-fullWidth-98"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-85"
      >
        Join
      </span>
      <span
        className="MuiTouchRipple-root-102"
      />
    </button>
  </div>
</div>
`;
